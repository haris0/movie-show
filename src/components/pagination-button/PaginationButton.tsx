import BackIcon from "../../assets/BackIcon";

interface PaginationButtonProps {
  currentPage: number;
  totalPage: number;
  onClickPageChange: (type: 'next' | 'back') => void;
}

const PaginationButton = ({
  currentPage,
  totalPage,
  onClickPageChange,
}: PaginationButtonProps) => {
  return (
    <div className="flex gap-2 justify-center mb-5">
      {currentPage > 1 && (
        <button 
          type="button" 
          className="text-gray-900 flex items-center gap-2 cursor-pointer w-fit justify-self-center bg-white border border-gray-300 focus:outline-none rounded-lg text-sm px-3 py-2.5"
          onClick={() => onClickPageChange('back')}
        >
          <BackIcon color="#000000" width={20}/>
          Back
        </button>
      )}
      {currentPage < totalPage && (
        <button 
          type="button" 
          className="text-gray-900 cursor-pointer flex items-center gap-2 w-fit justify-self-center bg-white border border-gray-300 focus:outline-none rounded-lg text-sm px-3 py-2.5"
          onClick={() => onClickPageChange('next')}
        >
          Next
          <BackIcon color="#000000" width={20} className="rotate-180"/>
        </button>
      )}
    </div>
  );
};

export default PaginationButton;
