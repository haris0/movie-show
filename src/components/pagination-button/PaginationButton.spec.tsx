import { render } from '@testing-library/react';
import PaginationButton from './PaginationButton';

const mockData = {
  currentPage: 2,
  totalPage: 10,
  onClickPageChange: () => undefined,
};

describe('PaginationButton Component', () => {
  it('should render correctly', () => {
    const { baseElement } = render(
      <PaginationButton {...mockData}/>,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
