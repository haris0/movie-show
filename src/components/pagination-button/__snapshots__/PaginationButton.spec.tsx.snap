// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PaginationButton Component should render correctly 1`] = `
<body>
  <div>
    <div
      class="flex gap-2 justify-center mb-5"
    >
      <button
        class="text-gray-900 flex items-center gap-2 cursor-pointer w-fit justify-self-center bg-white border border-gray-300 focus:outline-none rounded-lg text-sm px-3 py-2.5"
        type="button"
      >
        <svg
          fill="#000000"
          stroke="#000000"
          viewBox="0 0 1024 1024"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="SVGRepo_bgCarrier"
            stroke-width="0"
          />
          <g
            id="SVGRepo_tracerCarrier"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <g
            id="SVGRepo_iconCarrier"
          >
            <path
              d="M224 480h640a32 32 0 1 1 0 64H224a32 32 0 0 1 0-64z"
              fill="#000000"
            />
            <path
              d="m237.248 512 265.408 265.344a32 32 0 0 1-45.312 45.312l-288-288a32 32 0 0 1 0-45.312l288-288a32 32 0 1 1 45.312 45.312L237.248 512z"
              fill="#000000"
            />
          </g>
        </svg>
        Back
      </button>
      <button
        class="text-gray-900 cursor-pointer flex items-center gap-2 w-fit justify-self-center bg-white border border-gray-300 focus:outline-none rounded-lg text-sm px-3 py-2.5"
        type="button"
      >
        Next
        <svg
          class="rotate-180"
          fill="#000000"
          stroke="#000000"
          viewBox="0 0 1024 1024"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="SVGRepo_bgCarrier"
            stroke-width="0"
          />
          <g
            id="SVGRepo_tracerCarrier"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <g
            id="SVGRepo_iconCarrier"
          >
            <path
              d="M224 480h640a32 32 0 1 1 0 64H224a32 32 0 0 1 0-64z"
              fill="#000000"
            />
            <path
              d="m237.248 512 265.408 265.344a32 32 0 0 1-45.312 45.312l-288-288a32 32 0 0 1 0-45.312l288-288a32 32 0 1 1 45.312 45.312L237.248 512z"
              fill="#000000"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
</body>
`;
